### PR TITLE
Private registry

### PR DESCRIPTION
**Pull request by TerryHowe**
Currently, we use the registry for the bundle to install every package so that we can support private registries. This works for now since we currently only support everything from public ecr or a private registry. This change allows us to support:

1. private registries
2. sourceRegistry specified in the package custom resource
3. registry specified in the bundle
4. default registry will be the source registry of the bundle if nothing else is specified.

### CODE REVIEW

File: api/v1alpha1/packagebundlecontroller_types.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ type PackageBundleControllerSpec struct {
 	// +optional
 	ActiveBundle string `json:"activeBundle"`
 
+	// PrivateRegistry is the OCR address hosting the bundle.
```
Comment (lewisdiamond) [inline api/v1alpha1/packagebundlecontroller_types.go:67]: OCI + ECR = OCR?

File: pkg/packages/manager.go [hunk 2]
```diff
@@ -60,7 +61,9 @@ func processInstalling(mc *ManagerContext) bool {
 		mc.Log.Error(err, "Install failed")
 		return true
 	}
-	values[sourceRegistry] = mc.Source.Registry
+	if mc.Registry != "" && values[sourceRegistry] == "" {
```
Comment (junshun) [inline pkg/packages/manager.go:64]: Is there a scenario where both of these are set? I would image if Registry is set the user would want to use that regardless of sourceRegistry being there?

File: charts/eks-anywhere-packages/crds/crd.yaml [hunk 3]
```diff
@@ -210,6 +214,10 @@ spec:
                               schema:
                                 description: Schema is a base64 encoded, gzipped json schema used to validate package configurations
                                 type: string
+                              schema:
```
Comment (lewisdiamond) [inline charts/eks-anywhere-packages/crds/crd.yaml:217]: What's with this?

File: charts/eks-anywhere-packages/crds/crd.yaml [hunk 4]
```diff
@@ -300,6 +308,10 @@ spec:
                                   schema:
                                     description: Schema is a base64 encoded, gzipped json schema used to validate package configurations
                                     type: string
+                                  schema:
```
Comment (lewisdiamond) [inline charts/eks-anywhere-packages/crds/crd.yaml:311]: And this?

File: charts/eks-anywhere-packages/crds/crd.yaml [hunk 5]
```diff
@@ -321,7 +333,7 @@ spec:
                 enum:
                 - available
                 - ignored
-                - invalid version
```
Comment (lewisdiamond) [inline charts/eks-anywhere-packages/crds/crd.yaml:324]: Wasn't that already merged?

**Comment by lewisdiamond** [comment]
Can you explain what this feature is for?

File: api/v1alpha1/packagebundlecontroller_types.go (hunk 1 above)
Comment (TerryHowe) [reply api/v1alpha1/packagebundlecontroller_types.go:67]: copy pasta error from PackageBundleControllerSource

File: charts/eks-anywhere-packages/crds/crd.yaml (hunk 3 above)
Comment (TerryHowe) [reply charts/eks-anywhere-packages/crds/crd.yaml:217]: no idea why it generated that

File: pkg/packages/manager.go (hunk 2 above)
Comment (TerryHowe) [reply pkg/packages/manager.go:64]: So this `values[sourceRegistry]` comes from Package and that should override everything

Comment (junshun) [reply pkg/packages/manager.go:64]: So does that mean sourceRegistry from PackageBundleController spec does nothing?

File: charts/eks-anywhere-packages/crds/crd.yaml (hunk 5 above)
Comment (TerryHowe) [reply charts/eks-anywhere-packages/crds/crd.yaml:324]: uggh, missed that.

File: api/v1alpha1/packagebundlecontroller_types.go (hunk 1 above)
Comment (TerryHowe) [reply api/v1alpha1/packagebundlecontroller_types.go]: fixed

File: charts/eks-anywhere-packages/crds/crd.yaml (hunk 3 above)
Comment (TerryHowe) [reply charts/eks-anywhere-packages/crds/crd.yaml]: fixed

File: charts/eks-anywhere-packages/crds/crd.yaml (hunk 4 above)
Comment (TerryHowe) [reply charts/eks-anywhere-packages/crds/crd.yaml]: fixed

File: charts/eks-anywhere-packages/crds/crd.yaml (hunk 5 above)
Comment (TerryHowe) [reply charts/eks-anywhere-packages/crds/crd.yaml]: fixed

File: pkg/packages/manager.go (hunk 2 above)
Comment (TerryHowe) [reply pkg/packages/manager.go:64]: What should happen is
1. use `values[sourceRegistry]` which is from Package
2. use privateRegistry from PackageBundleController
3. use registry from bundle
4. use bundle registry from PackageBundleController

Comment (ewollesen) [reply pkg/packages/manager.go:64]: Is there a good way to get the comment above into the code in some (preferably) obvious and readable way?

Comment (TerryHowe) [reply pkg/packages/manager.go]: Good idea, added a method to ManagerContext

**Comment by TerryHowe** [comment]
> Can you explain what this feature is for?

Yes, updated the description

File: pkg/packages/manager.go (hunk 2 above)
Comment (TerryHowe) [reply pkg/packages/manager.go]: added some tests too

**Comment by lewisdiamond** [comment]
This is missing the helm CRDs update. Run `make helm-build`.

**Comment by jonahjon** [comment]
Does this mean we want to add back Bundle the field `api.BundlePackageSource.Registry`?

**Comment by TerryHowe** [comment]
> Does this mean we want to add back Bundle the field `api.BundlePackageSource.Registry`? Or should we continue to remove it from the bundle as we have been.

We should leave it out.

**Comment by TerryHowe** [comment]
> This is missing the helm CRDs update. Run `make helm-build`.

done

**Pull request merged**